source/common/tls/context_impl.cc

```diff
@@ -572,6 +584,24 @@ std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> ContextImpl::getPrivateKeyMe
   return providers;
 }
 
+void ContextImpl::updateCertStats() {
+  // Update CA cert expiration time.
+  cert_validator_->refreshCertStatsWithExpirationTime();
+
+  // Update TLS certs' expiration time.
+  for (Ssl::TlsContext& ctx : tls_contexts_) {
+    absl::optional<uint64_t> expiration_unix_time =
+        Utility::getExpirationUnixTime(ctx.cert_chain_.get());
+    if (expiration_unix_time.has_value()) {
+      absl::Status status = ctx.setExpirationOnCertStats(
```

Comment:
Can this fail? If not, make the function return void

Reply:
Sure let's update to void